design/tool.behavior.md

```diff
@@ -0,0 +1,95 @@
+#  Tool Behavior
+
+The tool behavior allows the user to assign this behavior to an actor so that the actor can be held and
+used.
+
+## Architecture
+
+Create a new behavior that is an `ToolBehavior` for being able to add to an actor, that when added will
+automatically enable grabbable on that actor.  This behavior will then allow the user to execute a primary 
+action to begin recording the transform changes over time.
+
+``` ts
+/**
+ * Tool behavior class containing the target behavior actions.
+ */
+export class ToolBehavior extends TargetBehavior {
+	private _holding: DiscreteAction = new DiscreteAction();
+	private _using: DiscreteAction = new DiscreteAction();
+
+	/** @inheritdoc */
+	public get behaviorType(): BehaviorType { return 'tool'; }
+
+	public get holding() { return this._holding; }
+	public get using() { return this._using; }
+
+	/**
+	 * Add a callback to the tool behavior to receive path updates for while the tool is being used
+	 */
+	public onUsingPathReceived((path: Transform[]) => void) {
+		// Path callback registration.
+	}
+
+	/**
+	 * Remove a callback to the tool behavior to receive path updates for while the tool is being used
+	 */
+	public offUsingPathReceived((path: Transform[]) => void) {
+		// Path callback de-registration.
+	}
+
+	/**
+	 * Add a holding handler to be called when the given hover state is triggered.
+	 * @param holdingState The holding state to fire the handler on.
+	 * @param handler The handler to call when the holding state is triggered.
+	 * @return This tool behavior.
+	 */
+	public onHolding(holdingState: 'grabbed' | 'dropped', handler: ActionHandler): this {
+		const actionState: ActionState = (holdingState === 'grabbed') ? 'started' : 'stopped';
+		this._holding.on(actionState, handler);
+		return this;
+	}
+
+	/**
+	 * Add a using handler to be called when the given hover state is triggered.
+	 * @param usingState The using state to fire the handler on.
+	 * @param handler The handler to call when the using state is triggered.
+	 * @return This tool behavior.
+	 */
+	public onUsing(usingState: 'started' | 'stopped', handler: ActionHandler): this {
+		const actionState: ActionState = (hoverState === 'started') ? 'started' : 'stopped';
+		this._using.on(actionState, handler);
+		return this;
+	}
+}
+```
+
+## Network
+
+The network messages will use the behavior payloads of SetBehavior and PerformAction payloads the same way all
+behaviors do.  In addition, there is an additional message that would come in the form of the path of the tool
+while it is being used.  This will be sent with a higher fidelity than the normal 10 hz transform updates, and 
+will be in the form of a collection of transform recordings.
+
+The message payload for the using path updates could look like the following:
+
+``` ts
+export type UsingToolPath = Payload & {
```

Comment:
Instead of this, I think we should update the actor transform subscription to support high-fidelity.

Reply:
If you want that to be a change to the update message, I am not against that.  This was added here under the assumption I had that we did not want to overload network traffic with a complete update patch just to implement this behavior.